File: aeongseo/BOJ/CPP/boj1863.cpp
/*** 1863. 스카이라인 쉬운거 ***/

#include<iostream>
#include<stack>
using namespace std;

int main() {
	int n, x, y;
	cin >> n;
	stack<int> s;

	int cnt = 0;
	for (int i = 0; i < n; i++) {
		cin >> x >> y;
		// 높이 감소할 때
		while (!s.empty() && s.top() > y) {
			s.pop();
			cnt++;
		}
		// 높이 증가할 때 (높이 감소에 사용된 y도 top의 값보다 크면 push)
		if (y > 0 && (s.empty() || s.top() < y)) {
			s.push(y);
		}
	}

	while (!s.empty()) {	// stack에 값이 남아있으면 pop, cnt 증가
		s.pop();
		cnt++;
	}

	cout << cnt;

	return 0;
}

File: bigsky/BOJ/Python/boj1863.py
# BOJ1863: 스카이라인 쉬운거
import sys
input = sys.stdin.readline

n = int(input())
stack = [0] # 숫자비교를 위해 0 넣음
cnt = 0

for _ in range(n):
    x, y = map(int, input().split())

    if stack[-1] < y: # top이 y보다 작다면 push(y)
        stack.append(y)

    elif stack[-1] > y: # top이 y보다 크다면 
        while stack[-1] > y: # y보다 큰 숫자는 차례로 모두 pop()
            stack.pop()
            cnt += 1
        if stack[-1] != y:
            stack.append(y)

while stack[-1] > 0: # 끝에는 모든 stack이 남지 않도록 pop()
    stack.pop()
    cnt += 1

print(cnt)
